src/osd/OSD.cc

```diff
@@ -7657,7 +7662,14 @@ vector<DaemonHealthMetric> OSD::get_health_metrics()
            << " currently "
            << op.state_string();
         lgeneric_subdout(cct,osd,20) << ss.str() << dendl;
-        clog->warn() << ss.str();
+        if (log_slowop) {
+          clog->warn() << ss.str();
+        } else {
+          std::string_view state = op.state_string();
```

Comment:
here is definitely a place to use 'auto'. As otherwise, if at any time in the future state_string() will return a string (as
suggested by its name), there will be a buggy silent conversion here.

Reply:
Removed code dealing with state_string().